src/cmd/services/m3coordinator/handler/placement/delete_all.go

```diff
@@ -39,16 +40,28 @@ const (
 type deleteAllHandler Handler
 
 // NewDeleteAllHandler returns a new instance of a placement delete all handler.
-func NewDeleteAllHandler(service placement.Service) http.Handler {
-	return &deleteAllHandler{service: service}
+func NewDeleteAllHandler(client clusterclient.Client, cfg config.Configuration) http.Handler {
+	return &deleteAllHandler{client: client, cfg: cfg}
 }
 
 func (h *deleteAllHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 	logger := logging.WithContext(ctx)
 
-	if err := h.service.Delete(); err != nil {
+	service, err := Service(h.client, h.cfg)
+	if err != nil {
+		handler.Error(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	if err := service.Delete(); err != nil {
 		logger.Error("unable to delete placement", zap.Any("error", err))
 		handler.Error(w, err, http.StatusInternalServerError)
 	}
+
+	json.NewEncoder(w).Encode(struct {
```

Comment:
I understand the rationale here, but I think it's a little weird that we don't return `Deleted: false` when we fail to delete here. I was thinking that the implicit `200 OK` response would be good enough here.

Reply:
We need some kind of output though I think, otherwise it's hard to know whether anything happened or not. 

I updated `handler.Error(...)` to return `{"error": "<the error>"}` so it should be pretty clean experience between the two.